### PR TITLE
Feature: Downloading multiple attachments

### PR DESCRIPTION
**Pull request by matteo1222**
## Change Summary

Implemented the feature - Multiselect for downloading attachments. ref: #335
Enable users to bulk download attachments of a column.

## Change type

- [x]  feat: (new feature for the user, not a new feature for build script)
- [ ]  fix: (bug fix for the user, not a fix to a build script)
- [ ]  docs: (changes to the documentation)
- [ ]  style: (formatting, missing semi colons, etc; no production code change)
- [ ]  refactor: (refactoring production code, eg. renaming a variable)
- [ ]  test: (adding missing tests, refactoring tests; no production code change)
- [ ]  chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

- Added a <downloadAttachmentsModal> component in the <moreAction> component. Users can select a field to download all attachments of the specified column.
- Added an API 'xcDownloadAttachments' in handleRequest in NcMetaMgr.ts. Used Archiver to create a zip file, stream add files, and pipe to response, which enables clients to download the zip file.
- It is implemented to be hidden in shared view for simplicity at the moment.

## Additional information / screenshots (optional)
https://user-images.githubusercontent.com/49985851/163179339-18308d87-7ac5-4d71-978e-5008aa4ab415.mov

## Anything for maintainers to be made aware of
I am new to this project and also new to open source, so I would love any feedback you have. Thank you so much!

### CODE REVIEW

File: packages/nc-gui/components/project/spreadsheet/components/moreActions.vue [hunk 1]
```diff
@@ -268,6 +291,27 @@ export default {
         this.$toast.error(e.message).goAway(3000)
       }
     },
+    async downloadAttachments({ selectedAttachmentField }) {
```
Comment (wingkwong) [inline packages/nc-gui/components/project/spreadsheet/components/moreActions.vue:294]: We don't do this anymore. We'd like to have a API for downloading attachments. Please check out our latest code and you may check `csvExport` as a reference.

File: packages/nocodb/src/lib/noco/meta/NcMetaMgr.ts [hunk 2]
```diff
@@ -1480,7 +1481,9 @@ export default class NcMetaMgr {
         case 'xcExportAsCsv':
           result = await this.xcExportAsCsv(args, req, res);
           break;
-
+        case 'xcDownloadAttachments':
```
Comment (wingkwong) [inline packages/nocodb/src/lib/noco/meta/NcMetaMgr.ts:1484]: We're not using NcMetaMgr.ts in the coming release. We put all new APIs under [here](https://github.com/nocodb/nocodb/tree/develop/packages/nocodb/src/lib/noco/meta/api).

File: packages/nc-gui/lang/en.json [hunk 3]
```diff
@@ -304,6 +304,7 @@
     "importExcel": "Import Excel",
     "downloadCSV": "Download as CSV",
     "uploadCSV": "Upload CSV",
+    "downloadAttachments": "Download Attachments",
```
Comment (wingkwong) [inline packages/nc-gui/lang/en.json:307]: To make it consistent, I'd suggest you to apply the same change to all files [here](https://github.com/nocodb/nocodb/tree/e251ed6f95e528b27fdee4d41ec5a19b01a74df9/packages/nc-gui/lang) with the same position. We'll update the master spreadsheet after that.

**Review by wingkwong — CHANGES_REQUESTED** [review]
Thanks for contributing. Great work! However, your local repo is quite outdated. We've changed a lot these days. Please pull the latest version, resolve the conflicts and check out my comments. If you need any help about our new design or if you have any doubts, please feel free to contact us in [Discord](https://discord.gg/5RgZmkW).

**Comment by matteo1222** [comment]
Got it. Thank you so much for the feedbacks!

**Comment by wingkwong** [comment]
Closing this PR as it's not compatible with the current design. Please feel free to contact us in Discord for discussion.

**Pull request closed without merging**